### PR TITLE
refactor: standardize performance collector patterns and error handling

### PR DESCRIPTION
**Pull request by jra3**
## Summary
- Standardized all performance collectors to follow consistent patterns established in CLAUDE.md
- Added proper error handling and path validation to all collector constructors
- Fixed interface consistency across all collectors

### First Commit: Update CLAUDE instructions with collector best practices
- Added comprehensive documentation for performance collector development patterns
- Documented testing methodology and requirements
- Established clear guidelines for error handling strategies

### Second Commit: Update existing collectors with best practices
- **Path Validation**: Added absolute path validation to all collectors (NetworkInfo, MemoryInfo, DiskInfo, TCP, Disk, CPUInfo, Kernel)
- **Constructor Standardization**: All constructors now return `(*Collector, error)` to handle validation errors
- **Interface Consistency**: Fixed all collectors to use `PointCollector` interface (previously some used base `Collector`)
- **Test Updates**: Added constructor tests for path validation to all collectors
- **Tool Updates**: Updated collector-bench tool to handle new error returns

### Path Validation Pattern
All collectors now validate their paths in constructors:
```go
if \!filepath.IsAbs(config.HostProcPath) {
    return nil, fmt.Errorf("HostProcPath must be an absolute path, got: %q", config.HostProcPath)
}
```

### Interface Standardization
All collectors that implement `Collect()` now properly use:
```go
var _ performance.PointCollector = (*CollectorName)(nil)
```

## Testing
- All collector tests pass ✅
- Linter passes with no errors ✅
- Added constructor tests for all collectors to verify path validation

## Follow-up
Created issue #54 to centralize the path validation logic that's currently duplicated across all collectors.

🤖 Generated with [Claude Code](https://claude.ai/code)

Co-Authored-By: Claude <noreply@anthropic.com>

### CODE REVIEW

File: CLAUDE.md [hunk 1]
```diff
@@ -245,30 +297,126 @@ Performance collectors follow a comprehensive testing pattern:
...
+   - Document error handling strategy
+3. **Create comprehensive tests** in `pkg/performance/collectors/your_collector_test.go`
+   - Use `collectors_test` package
+   - Include constructor tests
+   - Add helper functions following naming patterns
+   - Test edge cases and error scenarios
+4. **Update collector registry** if applicable
+5. **Run validation**:
+   ```bash
+   make test                    # Run tests
```
Comment (haq204) [inline CLAUDE.md:383]: add `make fmt` as well

File: CLAUDE.md [hunk 2]
```diff
@@ -245,30 +297,126 @@ Performance collectors follow a comprehensive testing pattern:
...
 
 5. **Testing Principles**
    - Don't test static properties (Name, RequiresRoot)
    - Focus on parsing logic and error handling
    - Use realistic test data from actual /proc files
-   - Test collectors in `collectors_test` package
+   - Test collectors in `collectors_test` package (external testing)
+   - Name test functions consistently: `TestXCollector_Constructor`, `TestXCollector_Collect`
+   - Use descriptive test names that explain the scenario
+   - Group related test scenarios in the same test function
```
Comment (haq204) [inline CLAUDE.md:358]: Also add don't add tests to check for interface validation. We have the compile time check for that

File: CLAUDE.md [hunk 3]
```diff
@@ -225,6 +261,22 @@ type CollectorCapabilities struct {
 }
 ```
 
+#### Error Handling Strategy
+Collectors must clearly distinguish between critical and optional data:
+- **Critical files**: Return error immediately if unavailable (e.g., /proc/loadavg for LoadCollector)
+- **Optional files**: Log warning and continue with graceful degradation (e.g., /sys/class/net/* metadata)
+- **Parse errors**: Handle based on field importance - critical fields cause errors, optional fields are logged
```
Comment (haq204) [inline CLAUDE.md:268]: Also add a note that collectors should never panic
